src/urllib3/response.py

```diff
@@ -618,6 +620,7 @@ def __init__(
 
         if hasattr(body, "read"):
             self._fp = body  # type: ignore[assignment]
+        self._sock_shutdown = sock_shutdown
```

Comment:
Should we set `self._sock_shutdown` to `None` when a response is closed to lose a reference to the socket?

I noticed that `shutdown` doesn't work after closing. Also, `self._fp.fp` (another reference to the socket) becomes `None` after closing, so maybe a still kept `self._sock_shutdown` can lead to memory leaks in some scenarios

Reply:
Great catch, fixed in [`f6d51ec` (#27)](https://github.com/pquentin/urllib3/pull/27/commits/f6d51ec05b89d66201d83719091fe1d750030a3f).

Reply:
Thanks!